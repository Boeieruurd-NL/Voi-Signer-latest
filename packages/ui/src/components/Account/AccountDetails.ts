import { FunctionalComponent } from "preact";
import { html } from 'htm/preact';
import { useContext } from 'preact/hooks';
import { route } from 'preact-router';
import qrcode from 'qrcode-generator';

import { StoreContext } from 'index'


const AccountDetails: FunctionalComponent = (props: any) => {
  const store:any = useContext(StoreContext);
  const { account, ledger } = props;

  const deleteAccount = () => {
    store.deleteAccount(ledger, account);
    route('/');
  }

  var typeNumber:TypeNumber = 4;
  var errorCorrectionLevel: ErrorCorrectionLevel = 'L';
  var qr = qrcode(typeNumber, errorCorrectionLevel);
  qr.addData(account.address);
  qr.make();
  let qrImg = qr.createDataURL(10, 1);

  return html`
    <strong>Address</strong>
    <p>${account.address}</p>

    <div style="text-align: center;">
      <img src="${qrImg}"
        style="padding: 0.5em;
          margin: 0.5em;
          border: 1px solid #9095AF;
          border-radius: 10px;"
        width="250"
        height="250"/>
    </div>

    <button
      class="button is-danger is-fullwidth"
      onClick=${deleteAccount}>
      Delete account!
    </button>

  `
};

export default AccountDetails;