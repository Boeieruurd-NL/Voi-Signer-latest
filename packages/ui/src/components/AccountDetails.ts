import { FunctionalComponent } from "preact";
import { html } from 'htm/preact';
import { useContext } from 'preact/hooks';
import { route } from 'preact-router';

import { StoreContext } from '../index'


const AccountDetails: FunctionalComponent = (props: any) => {
  const store:any = useContext(StoreContext);
  const { account, ledger } = props;

  const deleteAccount = () => {
    store.deleteAccount(ledger, account);
    route('/');
  }

  return html`
    <strong>Address</strong>
    <p>${account.address}</p>

    <strong>Mnemonic</strong>
    <p>${account.mnemonic}</p>

    <div class="panel-block">
      <button
        class="button is-danger is-fullwidth"
        onClick=${deleteAccount}>
        Delete account!
      </button>
    </div>
  `
};

export default AccountDetails;