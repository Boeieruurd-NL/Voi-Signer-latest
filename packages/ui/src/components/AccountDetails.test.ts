import { shallow } from 'enzyme';
import { html } from 'htm/preact';

import AccountDetails from './AccountDetails';

describe('AccountDetails', () => {
  it('should display initial count', () => {
    const account = {
      address: "",
      mnemonic: ""
    }
    const ledger = "TestNet";
    const component = shallow(html`<${AccountDetails}  account=${account} ledger ={ledger} />`);
    expect(component).toMatchSnapshot();
    // expect(true).toBe(true);
  });

  // it('should increment after "Increment" button is clicked', () => {
  //   const wrapper = mount(<Counter initialCount={5}/>);

  //   wrapper.find('button').simulate('click');

  //   expect(wrapper.text()).to.include('Current value: 6');
  // });
});