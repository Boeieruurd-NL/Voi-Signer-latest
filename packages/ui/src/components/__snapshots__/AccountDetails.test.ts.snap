// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetails should display initial count 1`] = `ShallowWrapper {}`;
